xc_fasm/fasm2frames.py

```diff
@@ -263,6 +264,31 @@ def check_for_pudc_b(set_feature):
         if missing_features:
             raise fasm_assembler.FasmLookupError('\n'.join(missing_features))
 
+    list_of_frames = list(assembler.frames_in_use)
+    list_of_frames.sort()
+
+    # Find CLB_IO_CLK/BRAM address ranges
+    clb_io_clk_list = list()
+    bram_list = list()
+    for addr in list_of_frames:
+        for info in assembler.grid.tileinfo.items():
+            block_types = info[1].bits.keys()
+            if BlockType.CLB_IO_CLK in block_types:
+                bits = info[1].bits[BlockType.CLB_IO_CLK]
+                if addr in range(bits.base_address,
+                                 bits.base_address + bits.frames):
+                    clb_io_clk_list.append(addr)
+            if BlockType.BLOCK_RAM in block_types:
+                bits = info[1].bits[BlockType.BLOCK_RAM]
+                if addr in range(bits.base_address,
+                                 bits.base_address + bits.frames):
+                    bram_list.append(addr)
+
+    clb_start_address  = None if clb_io_clk_list == [] else clb_io_clk_list[0]
```

Comment:
You're checking if the list is empty here, right? I think you could write: `clb_io_clk_list[0] if clb_io_clk_list else None`